randovania/game_connection/executor/dread_executor.py

```diff
@@ -50,9 +51,21 @@ class DreadExecutorToConnectorSignals(QObject):
 
 def get_bootstrapper_for(game: GameDescription) -> list[str]:
     all_code = [
+        # Pickup Locations
         textwrap.dedent("""
         Game.DoFile('actors/items/randomizer_powerup/scripts/randomizer_powerup.lua')
-        """).strip(),
+        RL.Pickups = {}
+        function RL.GetCollectedIndicesAndSend()
+            if not Scenario.CurrentScenarioID then return "not-in-game" end
+            local r,v,i,p = {},0,1,Game.GetPlayerBlackboardSectionName()
+            for _,t in ipairs(RL.Pickups) do
+                if Blackboard.GetProp(p,t) then v=v+i end
+                i=i*2;if i>=256 then table.insert(r,string.char(v));v=0;i=1 end
+            end
+            if i>1 then table.insert(r,string.char(v)) end
+            RL.SendIndices("locations:"..table.concat(r))
+        end""").strip(),
+        f"for i=1,{game.region_list.num_pickup_nodes} do RL.Pickups[i]='' end",
```

Comment:
we should move any substantial amounts of Lua like this into `.lua` files and load from there imo

Reply:
Hmm what would be the proper way for doing that? The code is send in chunks because of the buffer limit of 4096. It get blocks of `all_code` and concats them as long as they not exceed the limit.
If I just move this to one lua file, I have to make sure that it is still in some sort of blocks.
And also the dynamcially created parts from the region_list or resource_database of the game needs to be filled.

Reply:
1) just split it into multiple files that's nbd
2) you could use ODR's lua editing library to handle the codegen parts

Reply:
Tried to use `replace_lua_template` of odr's `lua_util`. Just realised the first parameter is always a file within odr's `Path(__file__).parent.joinpath("templates")` 😥

Reply:
ah, dang, just slap a `# FIXME` on it for now then

Reply:
Adressed the requested changes. Should be good now?